arwen/interface.go

```diff
@@ -17,13 +17,10 @@ type StateStack interface {
 
 // ArgumentsParser defines the functionality to parse transaction data into arguments and code for smart contracts
 type ArgumentsParser interface {
-	GetArguments() ([][]byte, error)
-	GetCode() ([]byte, error)
+	GetFunctionArguments() ([][]byte, error)
```

Comment:
It is an argument parser. No need to add into the function name. keep it simple

Reply:
It makes sense. Now the argument parser distinguishes between function arguments and constructor / deploy arguments. Before this, it still worked, but just because `GetFunction()` returned `parser.code`, which was horrible (by coincidence, the code and the function name were at the same position in the @argument string).

Reply:
Yes, now we need to differentiate between constructor and regular function arguments. Name should be explicit. Name should not be ambiguous. Name is simple :)